op-challenger/fault/agent_test.go

```diff
@@ -0,0 +1,31 @@
+package fault
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/fault/types"
+	"github.com/ethereum-optimism/optimism/op-node/testlog"
+)
+
+// TestShouldResolve tests the [Agent.ShouldResolve] resolution logic.
+func TestShouldResolve(t *testing.T) {
+	log := testlog.Logger(t, log.LvlCrit)
+
+	t.Run("AgreeWithProposedOutput", func(t *testing.T) {
+		agent := NewAgent(nil, 0, nil, nil, nil, true, log)
+		require.False(t, agent.ShouldResolve(context.Background(), types.GameStatus(2)))
+		require.True(t, agent.ShouldResolve(context.Background(), types.GameStatus(1)))
+		require.False(t, agent.ShouldResolve(context.Background(), types.GameStatus(0)))
+	})
+
+	t.Run("DisagreeWithProposedOutput", func(t *testing.T) {
+		agent := NewAgent(nil, 0, nil, nil, nil, false, log)
+		require.True(t, agent.ShouldResolve(context.Background(), types.GameStatus(2)))
+		require.False(t, agent.ShouldResolve(context.Background(), types.GameStatus(1)))
+		require.False(t, agent.ShouldResolve(context.Background(), types.GameStatus(0)))
```

Comment:
We should use the `GameStatusInProgress` etc constants instead of casting ints.

Reply:
Good point, thanks!